controllers/controllers/docker.go

```diff
@@ -0,0 +1,172 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	anywhere "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	dockerp "github.com/aws/eks-anywhere/pkg/providers/docker"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type docker struct {
+	client eksaClient
+}
+
+type dockerClusterSpec struct {
+	cluster          *anywhere.Cluster
+	spec             *cluster.Spec
+	datacenterConfig *anywhere.DockerDatacenterConfig
+}
+
+func (d *docker) ReconcileControlPlane(ctx context.Context, cluster *anywhere.Cluster) error {
+	spec, err := d.client.BuildClusterSpec(ctx, cluster)
+	if err != nil {
+		return err
+	}
+
+	// Generate CAPI CP yaml
+	controlPlaneSpec, err := generateControlPlaneCAPISpecForCreate(spec)
+	if err != nil {
+		return err
+	}
+
+	return createControlPlaneObjectsFromYamlManifest(ctx, d.client, cluster, controlPlaneSpec)
+}
+
+func createControlPlaneObjectsFromYamlManifest(ctx context.Context, client client.Client, cluster *anywhere.Cluster, yamlManifest []byte) error {
+	// Convert yaml CP spec to objects
+	objs, err := yamlToUnstructured(yamlManifest)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range objs {
+		obj.SetNamespace(cluster.Namespace)
+		// TODO: this is super hacky. We should probably do this in the template
+		//  Once we move to structs and generate them individually this should be way easier
+		if needsClusterLabelNameControlPlane(obj) {
```

Comment:
what's the reason not to mark all resources with eksa cluster label?

Reply:
No really a reason, we could do that. In fact it could be useful in the future to "list" all objects related to a cluster
This was just a way of being explicit and only set it where it was needed
If we are going to set it in all objects, we need to figure out if there are going to be objects shared between clusters and what do we do in that scenario